src/julia/generic.jl

```diff
@@ -181,6 +181,21 @@ Base.convert(::Type{Array}, xd::DistMatrix{T}) where {T} =
 
 Base.Array(xd::DistMatrix) = convert(Array, xd)
 
+# super slow, but handy?
+function Base.setindex!(A::DistMatrix,
```

Comment:
I think you might end up regretting this one. You'll be trading a hard error for very slow code where you'd have to identify generic fallbacks through profiling.